streams/writable-streams/start.js

```diff
@@ -93,15 +95,74 @@ promise_test(() => {
 }, 'underlying sink\'s write or close should not be invoked if the promise returned by start is rejected');
 
 promise_test(t => {
-  const rejection = { name: 'this is checked' };
   const ws = new WritableStream({
     start() {
       return {
-        then(onFulfilled, onRejected) { onRejected(rejection); }
+        then(onFulfilled, onRejected) { onRejected(error1); }
       };
     }
   });
-  return promise_rejects(t, rejection, ws.getWriter().closed, 'closed promise should be rejected');
+  return promise_rejects(t, error1, ws.getWriter().closed, 'closed promise should be rejected');
 }, 'returning a thenable from start() should work');
 
+promise_test(t => {
+  const ws = recordingWritableStream({
+    start(controller) {
+      controller.error();
+    }
+  });
+  return promise_rejects(t, new TypeError(), ws.getWriter().write('a'), 'write() should fail')
```

Comment:
Can you remind me why this is a TypeError instead of whatever was passed to controller.error()? In particular this confuses me with respect to the immediately-below test where error1 is the error.